### PR TITLE
Core & Internals: type annotate sqlalchemy models. #5578 

### PR DESCRIPTION
**Pull request by rcarpa**

### CODE REVIEW

**Comment by rcarpa** [comment]
The PR is built on top of https://github.com/rucio/rucio/pull/6252 and https://github.com/rucio/rucio/pull/6271

**Review by bari12 — APPROVED** [review]

File: lib/rucio/core/transfer.py [hunk 1]
```diff
@@ -1096,7 +1096,7 @@ def prepare_transfers(
             logger(logging.WARNING, '%s: all available sources were filtered', rws)
             continue
 
-        update_dict = {
+        update_dict: Dict[Any, Any] = {
```
Comment (dchristidis) [inline lib/rucio/core/transfer.py:1099]: How did this not raise an exception? `Dict` and `Any` are not imported under normal execution.

Comment (rcarpa) [reply lib/rucio/core/transfer.py:1099]: This is scary ; but I just debugged it and that code IS executed

Comment (dchristidis) [reply lib/rucio/core/transfer.py:1099]: I can change it in #6277 if it proves necessary since this line is modified, but I am baffled.

Comment (dchristidis) [reply lib/rucio/core/transfer.py:1099]: It would appear that this is working perfectly fine, though we can’t explain why. :man_shrugging:

Comment (rcarpa) [reply lib/rucio/core/transfer.py:1099]: Would still be nice if you fix it in #6277

Comment (dchristidis) [reply lib/rucio/core/transfer.py:1099]: From [PEP-526](https://peps.python.org/pep-0526/#runtime-effects-of-type-annotations):

> Annotating a local variable will cause the interpreter to treat it as a local, even if it was never assigned to. Annotations for local variables will not be evaluated:
> 
> ```
> def f():
>     x: NonexistentName  # No error.
> ```

**Pull request merged**